src/classes/RD_AddDonationsBTN_CTRL.cls

```diff
@@ -53,18 +53,22 @@ public with sharing class RD_AddDonationsBTN_CTRL {
     public PageReference buttonClick(){
         Savepoint sp = Database.setSavepoint();
         try {
+
+            if (!Schema.SObjectType.Opportunity.isCreateable()) { 
+                ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.INFO, System.Label.RD_ErrorAddDonationPermissionDenied));
+                return null;
+            }
+
             system.assert(sc.getId() != null);
```

Comment:
What do you think about cleaning up this method a bit while you are in here? Since we are adding an error message  with proper display maybe we can handle this scenario better as well?
Can we also fix some of our other best practices as far as capitalization / formatting, I believe we are sticking with only cleaning up the method and only what will not change the behavior but I'm still fuzzy on what we are allowed to refactor when working on the legacy code. Here is how this error looks now if we navigate to the page without an id. 
![screen shot 2018-10-15 at 11 48 37 am](https://user-images.githubusercontent.com/450473/46965487-a0f31580-d070-11e8-9648-a777a5048945.png)

Reply:
Where recurring donations are involved we should tread very carefully.

Reply:
Mostly left the method alone but did make a change so that the error is handled more elegantly when navigating without an id.